### ISSUE TITLE
Clarify strategy for SSH private keys 

### ISSUE THREAD

**Issue by bryanhuntesl** [GoogleContainerTools/kaniko#707]
If I need to use SSH private keys as part of my build process and I'd like to avoid baking them into images are accidentally pushing to repository - what should I do? 

I am aware of a `/kaniko` directory mount but it hasn't been clear to me from what I've read in terms of how dereference my ~/.ssh/id_rsa or ssh-agent socket file from there.

Is this supported functionality in kaniko? Docker has now got support for SSH secrets - http://blog.oddbit.com/post/2019-02-24-docker-build-learns-about-secr/

**Comment by bryanhuntesl** [comment]
kaniko allows mounts . closing.

**Comment by adrianf0** [comment]
@bryanhuntesl : Could you elaborate how did you manage to mount a directory to a docker image being currently built? I am struggling currently with passing _secrets_ to the build image build process as well.

**Comment by diclophis** [comment]
As I understand it, using kubermetes `volumes` and `volumeMounts` its possible to "layer" into the "build context" a directory contained ssh key material mounted from a kubernetes `secret` ... however... the part that is unclear is exactly what is the safest directory to use for this purpose? for example if I were to mount them in the incorrect location wouldn't kaniko just `COPY` the mounted bits into the built image?

TLDR: is there a specific directory we should use when mounting private key material into the build context for kaniko?
